api_client/python/timesketch_api_client/client.py

```diff
@@ -550,16 +550,12 @@ def list_sigma_rules(self, as_pandas=False):
             return pandas.DataFrame.from_records(response.get('objects'))
 
         for rule_dict in response['objects']:
-            rule_uuid = rule_dict.get('id')
-            title = rule_dict.get('title')
-            es_query = rule_dict.get('es_query')
-            file_name = rule_dict.get('file_name')
-            description = rule_dict.get('description')
-            file_relpath = rule_dict.get('file_relpath')
-            index_obj = sigma.Sigma(
-                rule_uuid, api=self, es_query=es_query, file_name=file_name,
-                title=title, description=description,
-                file_relpath=file_relpath)
+            if not rule_dict:
+                raise ValueError('No rules found.')
+
+            index_obj = sigma.Sigma(api=self)
+            for key, value in rule_dict.items():
+                index_obj.data[key] = value
```

Comment:
I would rather have this as 

```python
index_obj.set_value(key, value)
```

with the 

```python
def set_value(self, key, value):
  self._data[key] = value
```

or something like that.... or have this as 

```python
rule_uuid = rule_dict.get(....
....
index_obj = sigma.Sigma(api=self)
index_obj.from_manual(
  rule_uuid=rule_uuid, title=title, ....)
```

Reply:
also you don't even create `.data` in the object and you don't put that as an attribute... also data would overwrite the `resource.data` which is a property that returns `self.lazyload_data()`

Reply:
will go with the set_value as the numbers of params can change, so this seems to be the most generic way to do it then.